# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmOpIter.h
# Commit: 16fd9844f2d2
# Full Hash: 16fd9844f2d2029feaba3c864b1a0f830f9da235
# Author: Andy Wingo <wingo@igalia.com>
# Date: 2020-03-18 16:05:52
# Regressor Bug: 1620197
# File Overlap Count: 1
# Description:
#   Bug 1620197 - Enable multiple results from WebAssembly functions r=lth
#   
#   This patch enables multi-value calls and returns, adding some tests, and
#   conditionally disabling a couple expect-fail reftests that now pass.
#   
# ==============================================================================

diff -r 0d19968c99a7 -r 16fd9844f2d2 js/src/wasm/WasmOpIter.h
--- a/js/src/wasm/WasmOpIter.h	Wed Mar 18 13:39:50 2020 +0200
+++ b/js/src/wasm/WasmOpIter.h	Wed Mar 18 10:29:53 2020 +0000
@@ -2061,12 +2061,6 @@
 
   const FuncType& funcType = env_.types[*funcTypeIndex].funcType();
 
-  // FIXME: Remove this check when full multi-value function returns land.
-  // Bug 1585909.
-  if (funcType.results().length() > MaxFuncResults) {
-    return fail("too many returns in signature");
-  }
-
 #ifdef WASM_PRIVATE_REFTYPES
   if (env_.tables[*tableIndex].importedOrExported &&
       funcType.exposesTypeIndex()) {
@@ -2131,12 +2125,6 @@
 
   const FuncType& funcType = env_.types[*funcTypeIndex].funcType();
 
-  // FIXME: Remove this check when full multi-value function returns land.
-  // Bug 1585909.
-  if (funcType.results().length() > MaxFuncResults) {
-    return fail("too many returns in signature");
-  }
-
   if (!popCallArgs(funcType.args(), argValues)) {
     return false;
   }